File: src/main/scala/robosky/uplands/block/AwokenAzoteMushroomBlock.scala
package robosky.uplands.block

import net.fabricmc.fabric.api.tag.TagRegistry
import net.minecraft.block.{Block, BlockState, PlantBlock}
import net.minecraft.entity.EntityContext
import net.minecraft.tag.BlockTags
import net.minecraft.util.Identifier
import net.minecraft.util.math.BlockPos
import net.minecraft.util.shape.VoxelShape
import net.minecraft.world.BlockView
import robosky.uplands.UplandsBlockTags

object AwokenAzoteMushroomBlock {
  val SHAPE: VoxelShape = Block.createCuboidShape(5.0D, 0.0D, 5.0D, 11.0D, 6.0D, 11.0D)
}

class AwokenAzoteMushroomBlock(val settings: Block.Settings) extends PlantBlock(settings) {
  override def canPlantOnTop(blockState_1: BlockState, blockView_1: BlockView, blockPos_1: BlockPos): Boolean =
    blockState_1.matches(UplandsBlockTags.PlantableOn) ||
      blockState_1.matches(UplandsBlockTags.AzoteMushroomSpreadable)

  override def getOutlineShape(blockState_1: BlockState, blockView_1: BlockView, blockPos_1: BlockPos,
                               entityContext_1: EntityContext): VoxelShape = AwokenAzoteMushroomBlock.SHAPE
}

File: src/main/java/robosky/uplands/block/AwokenAzoteMushroomBlock.java
package robosky.uplands.block;

import net.minecraft.block.Block;
import net.minecraft.block.BlockState;
import net.minecraft.block.PlantBlock;
import net.minecraft.block.ShapeContext;
import net.minecraft.fluid.Fluids;
import net.minecraft.util.math.BlockPos;
import net.minecraft.util.shape.VoxelShape;
import net.minecraft.world.BlockView;

public class AwokenAzoteMushroomBlock extends PlantBlock {

    private static final VoxelShape SHAPE = Block.createCuboidShape(5.0D, 0.0D, 5.0D, 11.0D, 6.0D, 11.0D);

    public AwokenAzoteMushroomBlock(Settings settings) {
        super(settings);
    }

    @Override
    public VoxelShape getOutlineShape(BlockState blockState, BlockView blockView, BlockPos blockPos, ShapeContext entityContext) {
        return SHAPE;
    }

    @Override
    protected boolean canPlantOnTop(BlockState blockState, BlockView blockView, BlockPos blockPos) {
        return blockState.isIn(UplandsBlockTags.PLANTABLE_ON) || blockState.isIn(UplandsBlockTags.AZOTE_MUSHROOM_SPREADABLE);
    }
}
